app/Http/Controllers/CategoryController.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Http\Resources\CategoryResource;
+use App\Models\Category;
+use Illuminate\Http\Request;
+use App\Http\Requests\CategoryRequest;
+use Illuminate\Support\Facades\Validator;
+use Illuminate\Validation\Rule;
+
+
+class CategoryController extends Controller
+{
+    public function index(Request $request)
+    {
+        $data = [
+            'per_page' => 20,
```

Comment:
We don't want to hardcode per_page value, instead read it from the query param. If query param is not provided, the default value should be 20